### DISCUSSION TITLE
Help me. How to overlay the text of the currently playing audio track (meta data) on the video?

### DISCUSSION THREAD

**Discussion by voluburator** [savonet/liquidsoap#1952] [Q&A]
Help me. How to overlay the text of the currently playing audio track (meta datа) on the video? Help me. How to overlay the text of the currently playing audio track (meta date) on the video? Using my code example

set("init.daemon.pidfile.path", "/run/housemusicyoutube.pid")

set("log.file.path","/var/log/liquidsoap/housemusicyoutube.log")
set("log.level", 3)

audio = playlist(reload=1, mode="randomize", reload_mode="rounds", "/var/relaxmixer/housemusicyoutube/media/air")
audio = crossfade(smart=false, duration=6.50,fade_out=4.00,fade_in=4.00,audio)

audio = normalize(target = 0., window = 0.03, gain_min = -16., gain_max = 0., audio)
audio = compress.exponential(audio, mu = 1.0)

video = playlist(reload=1, mode="randomize", reload_mode="rounds", "/var/relaxmixer/housemusicyoutube/media/video")
stream = mux_video(video=video, audio)

enc = %ffmpeg(
  format="flv",
  %audio(codec="libmp3lame"),
 %video(codec="libx264")
)

key = "--------"
url = "rtmp://a.rtmp.youtube.com/live2/#{key}"
output.url(
  url=url,
  fallible=true,
  enc,
  stream)

**Comment by toots** [comment]
Hi!

I think the operator for that would be [video.add_text](https://www.liquidsoap.info/doc-dev/reference.html#video.add_text). The text itself is a string getter, which means that it can take a callback function..

Something like this:
```ruby
latest_title = ref("")

def title() =
  !latest_title
end

video = video.add_text(title, video, ....)
```

Then you can update `latest_title` at any point when a metadata comes through.

**Reply by voluburator** [reply]
Help me please. How to get metadata from the file being played (in the string: artist and track title) and add to a variable to pass it to video.add_text ? I am still not good at programming((
